app/controllers/api/imports.php

```diff
@@ -0,0 +1,403 @@
+    ->param('resources', [], new ArrayList(new WhiteList(Transfer::ALL_PUBLIC_RESOURCES)), 'List of resources to import')
+    ->param('endpoint', '', new URL(), "Source's Appwrite Endpoint")
+    ->param('projectId', '', new UID(), "Source's Project ID")
+    ->param('apiKey', '', new Text(512), "Source's API Key")
+    ->inject('response')
+    ->inject('dbForProject')
+    ->inject('project')
+    ->inject('user')
+    ->inject('events')
+    ->action(function (array $resources, string $endpoint, string $projectId, string $apiKey, Response $response, Database $dbForProject, Document $project, Document $user, Event $eventsInstance) {
+        $import = $dbForProject->createDocument('imports', new Document([
+            '$id' => ID::unique(),
+            'status' => 'pending',
+            'stage' => 'init',
+            'source' => json_encode([
+                'type' => 'appwrite',
+                'endpoint' => $endpoint,
+                'projectId' => $projectId,
+                'apiKey' => $apiKey,
+            ]),
+            'resources' => $resources,
+            'statusCounters' => '{}',
+            'resourceData' => "{}",
+            'errorData' => ""
+        ]));
+
+        $eventsInstance->setParam('importId', $import->getId());
+
+        // Trigger Transfer
+        $event = new Import();
+        $event
+            ->setImport($import)
+            ->setProject($project)
+            ->setUser($user)
+            ->trigger();
+
+        $response
+            ->setStatusCode(Response::STATUS_CODE_CREATED)
+            ->dynamic($import, Response::MODEL_IMPORT);
+    });
+
+App::post('/v1/imports/firebase')
+    ->groups(['api', 'imports'])
+    ->desc('Import Firebase Data')
+    ->label('scope', 'imports.write')
+    ->label('event', 'imports.create')
+    ->label('audits.event', 'import.create')
+    ->label('sdk.auth', [APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'imports')
+    ->label('sdk.method', 'importFirebase')
+    ->label('sdk.description', '/docs/references/imports/import-firebase.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_OK)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_IMPORT)
+    ->param('resources', [], new ArrayList(new WhiteList(Transfer::ALL_PUBLIC_RESOURCES)), 'List of resources to import')
+    ->param('serviceAccount', '', new Text(512), "Source's Service Account")
+    ->inject('response')
+    ->inject('dbForProject')
+    ->inject('project')
+    ->inject('user')
+    ->inject('events')
+    ->action(function (array $resources, string $serviceAccount, Response $response, Database $dbForProject, Document $project, Document $user, Event $eventsInstance) {
+        $import = $dbForProject->createDocument('imports', new Document([
+            '$id' => ID::unique(),
+            'status' => 'pending',
+            'stage' => 'init',
+            'source' => json_encode([
```

Comment:
No need to use json_encode here, we already have a database filter we can auto apply for this. Also, wouldn't it make more sense to call this attribute credentials and save the source type in a separate attribute?